### DISCUSSION TITLE
About Data TTL in mysql

### DISCUSSION THREAD

**Discussion by Alexios-cd** [apache/skywalking#10629] [Q&A] [answered]
I am using skywalking 9.4.0 and using MySQL for storage. According to the official documentation（ https://skywalking.apache.org/docs/main/next/en/setup/backend/ttl/#ttl ），There are only two configuration items: recordDataTTL and metricsDataTTL. I wonder if there are configurations such as minuteMetricsDataTTL, hourMetricsDataTTL, dayMetricsDataTTL, and monthMetricsDataTTL. Do I need to regularly clean up historical data in MySQL?

According to online information, the previous historical version has these configurations, but the new version 9.4.0 does not have these configurations. What should I do if I want to clean up the data in MySQL on time?

**Comment by wankai123** [comment]
These  2 TTL configurations are for different data types as the document mentioned. OAP will clean up all expired data which out of TTL, including min/hour/day.

**Reply by Alexios-cd** [reply]
I want to know the cleaning time for minute data, hour data, and day data. By default
recordDataTTL: ${SW_CORE_RECORD_DATA_TTL:3} # Unit is day
metricsDataTTL: ${SW_CORE_METRICS_DATA_TTL:7} # Unit is day
How many days will minute data, hour data, and day data be cleaned up?
What data cleansing mechanisms does the recordDataTTL contain?
What data cleansing mechanisms does the metricsDataTTL include?

**Reply by wankai123** [reply]
>recordDataTTL: ${SW_CORE_RECORD_DATA_TTL:3} # Unit is day
>metricsDataTTL: ${SW_CORE_METRICS_DATA_TTL:7} # Unit is day
>How many days will minute data, hour data, and day data be cleaned up?

record data 7 days and metrics data 3 days according to your configuration.

**Reply by wankai123** [reply]
no matter minute data, hour data or day data

**Reply by Alexios-cd** [reply]
What is recordData?
What is metricsData?
What is the difference between recordData and metricsData? Can you give me an example?
According to the documentation, I don't know which data will be cleaned up after 3 days, and which data will be cleaned up after 7 days?

**Reply by wu-sheng** [reply]
Isn't doc clearly mentioning the data types? Why repeat the questions?

**Reply by wu-sheng** [reply, marked as answer]
Read the docs again. Observe the data. 
That is just a simple action you can take. Also, the source codes have been opened for you.
Taking action is always better than seeking an answer repeatedly in the text. Try to do something.
